presto-main/src/test/java/io/prestosql/memory/TestLocalMemoryManager.java

```diff
@@ -17,63 +17,66 @@
 import org.testng.annotations.Test;
 
 import static io.airlift.units.DataSize.Unit.GIGABYTE;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.testng.Assert.assertEquals;
 import static org.testng.Assert.assertFalse;
 import static org.testng.Assert.assertTrue;
 
 public class TestLocalMemoryManager
 {
     @Test
-    public void testReservedMemoryDisabled()
+    public void testReservedMemoryPoolDisabled()
     {
         NodeMemoryConfig config = new NodeMemoryConfig()
                 .setReservedPoolEnabled(false)
                 .setHeapHeadroom(new DataSize(10, GIGABYTE))
                 .setMaxQueryMemoryPerNode(new DataSize(20, GIGABYTE))
                 .setMaxQueryTotalMemoryPerNode(new DataSize(20, GIGABYTE));
 
-        LocalMemoryManager localMemoryManager = new LocalMemoryManager(config,
-                new DataSize(60, GIGABYTE).toBytes());
+        LocalMemoryManager localMemoryManager = new LocalMemoryManager(config, new DataSize(60, GIGABYTE).toBytes());
         assertFalse(localMemoryManager.getReservedPool().isPresent());
         assertEquals(localMemoryManager.getPools().size(), 1);
     }
 
     @Test
-    public void testReservedMemoryEnabled()
+    public void testReservedMemoryPoolEnabled()
     {
         NodeMemoryConfig config = new NodeMemoryConfig()
                 .setHeapHeadroom(new DataSize(10, GIGABYTE))
                 .setMaxQueryMemoryPerNode(new DataSize(20, GIGABYTE))
                 .setMaxQueryTotalMemoryPerNode(new DataSize(20, GIGABYTE));
 
-        LocalMemoryManager localMemoryManager = new LocalMemoryManager(config,
-                new DataSize(60, GIGABYTE).toBytes());
+        LocalMemoryManager localMemoryManager = new LocalMemoryManager(config, new DataSize(60, GIGABYTE).toBytes());
         assertTrue(localMemoryManager.getReservedPool().isPresent());
         assertEquals(localMemoryManager.getPools().size(), 2);
     }
 
-    @Test(expectedExceptions = IllegalArgumentException.class)
-    public void testMaxQueryMemoryPerNodeBiggerQueryTotalMemoryPerNode()
+    @Test
+    public void testMaxQueryMemoryPerNodeBiggerThanQueryTotalMemoryPerNode()
     {
         NodeMemoryConfig config = new NodeMemoryConfig()
                 .setHeapHeadroom(new DataSize(10, GIGABYTE))
                 .setMaxQueryMemoryPerNode(new DataSize(200, GIGABYTE))
                 .setMaxQueryTotalMemoryPerNode(new DataSize(20, GIGABYTE));
 
-        new LocalMemoryManager(config, new DataSize(60, GIGABYTE).toBytes());
+        assertThatThrownBy(() -> new LocalMemoryManager(config, new DataSize(60, GIGABYTE).toBytes()))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageMatching("Max query memory per node \\(query.max-memory-per-node\\)" +
```

Comment:
Technically `.` should be escaped. Though I think you can simplify this using `hasMessageStartingWith()`